### PR TITLE
feat(ces/alarmrule): support event monitoring in alarmrule

### LINKED ISSUE #1952: [CES]plan to refactor ces_alarmrule
1. support multiple policies in one rule
2. support more alarm types
3. support more resources attached to one rule(currently supports 3)
4. more detailed descriptions of parameters `namespace`, `metric` and `condition`

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

support event monitoring in alarmrule

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1952 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support event monitoring in alarmrule
2. change 'dimensions' to optional and remove the limit on the number of 'dimensions'
3. update doc of alarmrule
4. move ces_alarmrule resource to services path
5. due to API problem, multiple policies in one rule is not supported
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ces' TESTARGS='-run TestAccCESAlarmRule_sysEvent'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ces -v -run TestAccCESAlarmRule_sysEvent -timeout 360m -parallel 4
=== RUN   TestAccCESAlarmRule_sysEvent
=== PAUSE TestAccCESAlarmRule_sysEvent
=== CONT  TestAccCESAlarmRule_sysEvent
--- PASS: TestAccCESAlarmRule_sysEvent (148.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ces       149.095s
```

### CODE REVIEW

File: docs/resources/ces_alarmrule.md [hunk 1]
```diff
@@ -58,6 +89,9 @@ The following arguments are supported:
   which indicates *critical*, *major*, *minor*, and *informational*, respectively.
   The default value is 2.
 
+* `alarm_type` - (Optional, String) Specifies the alarm type. The value can be **EVENT.SYS**, **EVENT.CUSTOM**
```
Comment (niuzhenguo) [inline docs/resources/ces_alarmrule.md:92]: better to note that which alarm_type is the default value.

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go [hunk 2]
```diff
@@ -94,8 +97,7 @@ func ResourceAlarmRule() *schema.Resource {
 
 						"dimensions": {
 							Type:     schema.TypeList,
-							Required: true,
-							MaxItems: 3,
```
Comment (niuzhenguo) [inline huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go:98]: the MaxItemss limitation is not needed now?

Comment (Jason-Zhang9309) [reply huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go:98]: yes, there is not limit in the API and console

File: docs/resources/ces_alarmrule.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/ces_alarmrule.md:92]: sure, fixed

File: huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go [hunk 3]
```diff
@@ -178,6 +180,12 @@ func ResourceAlarmRule() *schema.Resource {
 				ValidateFunc: validation.IntBetween(1, 4),
 			},
 
+			"alarm_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"EVENT.SYS", "EVENT.CUSTOM", "MULTI_INSTANCE"}, false),
```
Comment (ShiChangkuo) [inline huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go:187]: missing computed?

Comment (Jason-Zhang9309) [reply huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go:187]: fixed

**Pull request merged**